pkg/generator/generator.go

```diff
@@ -53,205 +57,239 @@ type keyGenConfig struct {
 
 // GenKeys load secrets from a secret manager or generate them and save to a secret manager
 // GenKeys generates keys until there's an error or a dependency that can't be set.
-func GenKeys(config *GenConfig, keys []*v1alpha1.KeyConfig) error {
+func (g *GenConfig) GenKeys(ctx context.Context) error {
 	// A queue that will allow re-processing of self referencing keys that exist
-	queue, enqueuer := keyQueue(keys)
-	// generate until queue done
-	for key := range queue {
-		log.V(1).Info("reconciling secret key", "secret_name", secretReq.Name,
-			"data_key", key.Name, "secret_type", string(key.Type))
-
-		keyGenerator, err := generator.NewKeyGenerator(secretReq.Name, instance.Namespace, key, &instance.Spec.AppConfig)
-		if err != nil {
-			log.V(0).Info("error routing secret key type",
-				"secret_name", secretReq.Name,
+	g.Log = g.Log.WithValues("secret_name", g.SecObject.Name, "namespace", g.Namespace)
+	requeue, enquerer := makeKeyQueue(g.KeysToGen)
+	for {
+		// all work completed, exit
+		if len(requeue) == 0 {
+			g.Log.V(1).Info("finished preparing secret")
+			break
+		}
+		// make queue from requeue
+		queue := append(requeue[:0:0], requeue...)
+		// reset re-work queue
+		requeue = []*v1alpha1.KeyConfig{}
+		// process all keys in queue
+		for _, key := range queue {
+			log := g.Log.WithValues(
 				"data_key", key.Name,
 				"secret_type", string(key.Type))
-			return err
-		}
-		empty, err := keyGenerator.SecretManagerHasData(ctx)
-		if err != nil {
-			return err
-		}
-		if empty {
-			// load dependencies.
-			// missing secrets and keys are treated are errors and should stop generation
-			// self referencing deps can be resolved by re-running again later
-			reEnqueForKey, err := keyGenerator.ConfigureDependencies(ctx, rclient)
-			// check for NotFoundErr, for retry
+			log.V(1).Info("key started")
```

Comment:
key generation started

Reply:
more like _key processing started_. At this point, we don't know if the key is going to be loaded form a secret manager or generated